src/runtime-rs/crates/runtimes/src/tracer.rs

```diff
@@ -63,85 +59,88 @@ impl KataTracer {
     fn enabled(&self) -> bool {
         self.enabled
     }
-}
 
-/// Call when the tracing is enabled (set in toml configuration file)
-/// This setup the subscriber, which maintains the span's information, to global and
-/// inside KATA_TRACER.
-///
-/// Note that the span will be noop(not collected) if a invalid subscriber is set
-pub fn trace_setup(
-    sid: &str,
-    jaeger_endpoint: &str,
-    jaeger_username: &str,
-    jaeger_password: &str,
-) -> Result<()> {
-    let mut kt = KATA_TRACER.lock().unwrap();
-
-    // If varify jaeger config returns an error, it means that the tracing should not be enabled
-    let endpoint = verify_jaeger_config(jaeger_endpoint, jaeger_username, jaeger_password)?;
-
-    // enable tracing
-    kt.enable();
-
-    // derive a subscriber to collect span info
-    let tracer = opentelemetry_jaeger::new_collector_pipeline()
-        .with_service_name(format!("kata-sb-{}", &sid[0..min(8, sid.len())]))
-        .with_endpoint(endpoint)
-        .with_username(jaeger_username)
-        .with_password(jaeger_password)
-        .with_hyper()
-        .install_batch(Tokio)?;
-
-    let layer = tracing_opentelemetry::layer().with_tracer(tracer);
-
-    let sub = Registry::default().with(layer);
-
-    // we use Arc to let global subscriber and katatracer to SHARE the SAME subscriber
-    // this is for record the global subscriber into a global variable KATA_TRACER for more usages
-    let subscriber = Arc::new(sub);
-    tracing::subscriber::set_global_default(subscriber.clone())?;
-    kt.subscriber = subscriber;
-
-    info!(sl!(), "Tracing enabled successfully");
-    Ok(())
-}
+    /// Call when the tracing is enabled (set in toml configuration file)
+    /// This setup the subscriber, which maintains the span's information, to global and
+    /// inside KATA_TRACER.
+    ///
+    /// Note that the span will be noop(not collected) if a invalid subscriber is set
+    pub fn trace_setup(
+        &mut self,
+        sid: &str,
+        jaeger_endpoint: &str,
+        jaeger_username: &str,
+        jaeger_password: &str,
+    ) -> Result<()> {
+        // If varify jaeger config returns an error, it means that the tracing should not be enabled
+        let endpoint = verify_jaeger_config(jaeger_endpoint, jaeger_username, jaeger_password)?;
+
+        // derive a subscriber to collect span info
+        let tracer = opentelemetry_jaeger::new_collector_pipeline()
+            .with_service_name(format!("kata-sb-{}", &sid[0..min(8, sid.len())]))
```

Comment:
Is 8 bits sufficient to distinguish sandboxes, especially for business scenarios with many instances.

Reply:
We choose the first 8 characters of sid as unique id here. I think it is enough.

Reply:
we could surely use more, but eight characters make it already very rarely to conflict.

prob: (1/36**8) for two of them to collide.